src/logic/outfits.ts

```diff
@@ -54,3 +49,86 @@ export async function getOutfits(
     }
   }
 }
+
+function isOwnedOutfit(outfit: Outfit, ownedWearables: OnChainWearable[]) {
+  const parsedWearables = outfit.wearables.map((wearable) => splitUrnAndTokenId(wearable))
+  return parsedWearables.every((wearable) => {
+    if (wearable.urn.includes('off-chain')) {
+      return true
+    }
+
+    const matchingOwnedWearable = ownedWearables.find(
+      (ownedWearable) =>
+        ownedWearable.urn === wearable.urn &&
+        (!wearable.tokenId || ownedWearable.individualData.some((itemData) => itemData.tokenId === wearable.tokenId))
+    )
+
+    return !!matchingOwnedWearable
+  })
+}
+
+function parseValidWearablesAndFilterInvalidOnes(
+  wearablesUrn: string[],
+  ownedWearables: OnChainWearable[],
+  shouldExtendWearables: boolean
+): string[] {
+  const wearablesUrnToReturn: string[] = []
+
+  for (const wearable of wearablesUrn) {
+    if (wearable.includes('base-avatars')) {
+      wearablesUrnToReturn.push(wearable)
+      continue
+    }
+
+    const { urn, tokenId } = splitUrnAndTokenId(wearable)
+
+    const matchingOwnedWearable = ownedWearables.find(
+      (ownedWearable) =>
+        ownedWearable.urn === urn &&
+        (!tokenId || ownedWearable.individualData.find((itemData) => itemData.tokenId === tokenId))
+    )
+
+    if (!matchingOwnedWearable) {
+      continue
+    }
+
+    wearablesUrnToReturn.push(
+      shouldExtendWearables
+        ? `${matchingOwnedWearable.urn}:${tokenId ? tokenId : matchingOwnedWearable.individualData[0].tokenId}`
+        : matchingOwnedWearable.urn
+    )
+  }
+
+  return wearablesUrnToReturn
+}
+
+async function filterOutfitsWithoutCompleteOwnership(
+  components: Pick<AppComponents, 'wearablesFetcher' | 'config'>,
+  outfits: Outfits,
+  owner: string
+): Promise<Outfits> {
+  const { config, wearablesFetcher } = components
+  const ensureERC721 = (await config.getString('ENSURE_ERC_721')) === 'true'
+  const ownedWearables: OnChainWearable[] = await wearablesFetcher.fetchOwnedElements(owner)
+
+  const outfitsToReturn: { slot: number; outfit: Outfit }[] = []
+
+  for (const outfit of outfits.outfits) {
+    if (!isOwnedOutfit(outfit.outfit, ownedWearables)) {
+      continue
+    }
+
+    const parsedOutfitsWearables = parseValidWearablesAndFilterInvalidOnes(
+      outfit.outfit.wearables,
+      ownedWearables,
+      ensureERC721
+    )
+
+    outfitsToReturn.push({ ...outfit, outfit: { ...outfit.outfit, wearables: parsedOutfitsWearables } })
+  }
+
+  return {
+    ...outfits,
+    outfits: outfitsToReturn
```

Comment:
I noticed only outfits are used, maybe we should only return outfitsToReturn?

Reply:
Just pushed a commit with this change! Nice catch